--- a/cooking_recipe.cmake
+++ b/cooking_recipe.cmake
@@ -283,9 +283,12 @@ cooking_ingredient (dpdk
 
 cooking_ingredient (fmt
   EXTERNAL_PROJECT_ARGS
-    URL https://github.com/fmtlib/fmt/archive/5.2.1.tar.gz
-    URL_MD5 eaf6e3c1b2f4695b9a612cedf17b509d
+    SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/fmt
   CMAKE_ARGS
+    -DCMAKE_C_COMPILER=gcc-9
+    -DCMAKE_CXX_COMPILER=g++-9
+    -DCMAKE_C_FLAGS=-fPIC
+    -DCMAKE_CXX_FLAGS=-fPIC
     -DFMT_DOC=OFF
     -DFMT_TEST=OFF)
 
